p2p/src/peer.rs

```diff
@@ -374,67 +371,75 @@ mod run {
 mod state {
     //! Module for peer stages.
 
-    use iroha_crypto::PublicKey;
-    use iroha_primitives::const_vec::ConstVec;
+    use iroha_crypto::{KeyPair, PublicKey, Signature};
+    use iroha_primitives::{addr::SocketAddr, const_vec::ConstVec};
 
     use super::{cryptographer::Cryptographer, *};
 
     /// Peer that is connecting. This is the initial stage of a new
     /// outgoing peer.
     pub(super) struct Connecting {
-        pub peer_id: PeerId,
+        pub peer_addr: SocketAddr,
+        pub key_pair: KeyPair,
         pub connection_id: ConnectionId,
     }
 
     impl Connecting {
         pub(super) async fn connect_to(
             Self {
-                peer_id,
+                peer_addr,
+                key_pair,
                 connection_id,
             }: Self,
         ) -> Result<ConnectedTo, crate::Error> {
-            let stream = TcpStream::connect(peer_id.address.to_string()).await?;
+            let stream = TcpStream::connect(peer_addr.to_string()).await?;
             let connection = Connection::new(connection_id, stream);
             Ok(ConnectedTo {
-                peer_id,
+                peer_addr,
+                key_pair,
                 connection,
             })
         }
     }
 
     /// Peer that is being connected to.
     pub(super) struct ConnectedTo {
-        peer_id: PeerId,
+        peer_addr: SocketAddr,
+        key_pair: KeyPair,
         connection: Connection,
     }
 
     impl ConnectedTo {
         pub(super) async fn send_client_hello<K: Kex, E: Enc>(
             Self {
-                peer_id,
+                peer_addr,
+                key_pair,
                 mut connection,
             }: Self,
         ) -> Result<SendKey<E>, crate::Error> {
             let key_exchange = K::new();
-            let (local_public_key, local_private_key) = key_exchange.keypair(None)?;
-            let (algorithm, local_public_key_raw) = local_public_key.into_raw();
+            let (local_session_public_key, local_private_key) = key_exchange.keypair(None)?;
```

Comment:
The `local_private_key` is also not the real peer private key, but just used for key exchange, so the `local_session_public_key` and `local_private_key` are now somewhat inconsistently named..

Maybe name them `kx_local_public_key` and `kx_local_private_key`? Or shorten them to `kx_local_pk` and `kx_local_sc`? Same for the remote PK

Reply:
good point, this was confusing me

Reply:
Sorry, I made a typo, it should've been `kx_local_sk`, not `kx_local_sc` :sweat: